src/main/resources/Macros/MapMacro/Code/LeafletMap.xml

```diff
@@ -175,10 +176,14 @@
 
       var marker = L.marker(latLng, {icon: icon}).addTo(mymap);
       marker.on('click', onClick);
+      if(info != '')
+          marker.bindPopup(info);
       
       function onClick() {
-        var url = "https://www.openstreetmap.org/directions?engine=graphhopper_foot&amp;to="+ latLng+ "#map=19/"+ latLng[0]+"/"+ latLng[1];
-        window.open(url,'_blank');
+        if(info == ''){
```

Comment:
Please use a space before and after the parenthesis: we use the same JavaScript code style principles for this as in Java, see also [XWiki Java Code Style](https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/).